### PR TITLE
fix(vc): validate decoded credential shape in parseJwtCredential

### PR DESCRIPTION
**Pull request by venables**
## Summary

Follow-up to #113. Addresses the unchecked-cast review comment left on #110: `parseJwtCredential` cast `verifyCredential`'s result to `Verifiable<W3CCredential>` without validating it.

`verifyCredential` (did-jwt-vc) returns its own credential shape. `parseJwtCredential` now validates that shape conforms to our `W3CCredential` (via the existing `isCredential` / `credentialSchema`) before returning it, throwing `InvalidCredentialError` on a divergent shape rather than silently casting. Since `parseJwtCredential` is the single decode point used by `verifyProof` / `verifyParsedCredential` / `verifyPaymentReceipt`, this hardens the whole verification chain.

## Changes

- `packages/vc/src/verification/parse-jwt-credential.ts` — validate `result.verifiableCredential` with `isCredential` before returning.
- `packages/vc/src/verification/parse-jwt-credential.test.ts` — regression test for a non-conforming decoder result (delegating `did-jwt-vc` mock so the existing real-signing test still exercises the real implementation).

## Verification

- Legitimate did-jwt-vc output is unaffected (the real-signing parse test still passes).
- `@agentcommercekit/vc` 49/49 and `@agentcommercekit/ack-pay` 34/34 pass; full `pnpm run check` is green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Improved validation of JWT credentials by verifying the decoded credential’s structure before processing.
  * Updated proof verification to preserve `InvalidCredentialError` when credential shape validation fails, and treat other failures as `InvalidProofError`.
* **Tests**
  * Expanded `parseJwtCredential` and `verifyProof` test coverage for edge cases and additional credential payload fields.
* **Documentation**
  * Added a changeset noting that `parseJwtCredential` now validates credential shape instead of relying on unchecked casting.

### CODE REVIEW

File: packages/vc/src/verification/parse-jwt-credential.test.ts [hunk 1]
```diff
@@ -59,3 +68,16 @@ test("verifyCredentialJwt should throw for invalid credential", async () => {
...
+
+  // Simulate did-jwt-vc returning a shape that diverges from W3CCredential
+  vi.mocked(verifyCredential).mockResolvedValueOnce({
+    verifiableCredential: { not: "a credential" },
+  } as unknown as Awaited<ReturnType<typeof verifyCredential>>)
+
+  await expect(parseJwtCredential("a.b.c", resolver)).rejects.toThrow(
+    InvalidCredentialError,
+  )
+})
```
[the remark this answers is not in the document]
Comment (venables) [reply packages/vc/src/verification/parse-jwt-credential.test.ts]: Fixed in eaf5ccf — converted this file's tests to `it(...)` to match the rest of the suite and the AGENTS.md convention.
